examples/micropython/main.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       micropython example application
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "thread.h"
+
+#include "micropython.h"
+#include "py/stackctrl.h"
+#include "lib/utils/pyexec.h"
+
+#include "blob/boot.py.h"
+
+static char mp_heap[MP_RIOT_HEAPSIZE];
+
+int main(void)
+{
+    int coldboot = 1;
+
+    /* let MicroPython know the top of this thread's stack */
+    uint32_t stack_dummy;
+    mp_stack_set_top((char*)&stack_dummy);
+
+    /* Make MicroPython's stack limit somewhat smaller than actual stack limit */
+    mp_stack_set_limit(THREAD_STACKSIZE_MAIN - 128);
```

Comment:
Maybe add a define for this 'somewhat smaller', maybe something along the lines of `MICROPYTHON_STACKSIZE_REDUCTION`.

Reply:
ok, added to micropython.h